src/logger.js

```diff
@@ -0,0 +1,55 @@
+/* eslint no-console:0 */
+
+import chalk from 'chalk';
+import {isFunction, isNil} from 'lodash';
+
+function log(message, userFn, colorFn) {
+  if(!isNil(userFn) && !isFunction(userFn)) {
+    throw new TypeError('Extensions to knex logger must be functions!');
+  }
+
+  if(isFunction(userFn)) {
+    userFn(message);
+    return;
+  }
+
+  console.log(colorFn ? colorFn(message) : message);
+}
+
+class Logger {
+  constructor(config) {
+    const {
+      log: {
+        debug,
+        warn,
+        error,
+        deprecate
+      } = {}
+    } = config;
+
+    this._debug     = debug;
+    this._warn      = warn;
+    this._error     = error;
+    this._deprecate = deprecate;
+  }
+
+  debug(message) {
```

Comment:
this is going to swallow any parameters passed to the logger (e. g. error instance). Using 'arguments' instead is going to be much more robust.